From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Mon, 27 Nov 2023 20:40:28 +0800
Subject: [PATCH] arm64: fix abi change caused by ILP32

hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I8JVJ3
CVE: NA

----------------------------------------

One of the ILP32 patchset rename 'compat_user_mode' and
'compat_thumb_mode' to 'a32_user_mode' and 'a32_user_mode'. But these
two macros are used in some opensource userspace application. To keep
compatibility, we redefine these two macros.

Fixes: 23b2f00 ("arm64: rename functions that reference compat term")
Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: liwei <liwei391@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Jinjie Ruan <ruanjinjie@huawei.com>

(cherry picked from openEuler OLK-6.6 commit 8c1436bd1150537f807e483808224bd9de2121a7)
---
 arch/arm64/include/asm/ptrace.h | 4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/arm64/include/asm/ptrace.h
+++ b/arch/arm64/include/asm/ptrace.h
@@ -201,6 +201,8 @@ static inline void forget_syscall(struct
 #define a32_thumb_mode(regs) (0)
 #endif
 
+#define compat_thumb_mode(regs) a32_thumb_mode(regs)
+
 #define user_mode(regs)	\
 	(((regs)->pstate & PSR_MODE_MASK) == PSR_MODE_EL0t)
 
@@ -208,6 +210,8 @@ static inline void forget_syscall(struct
 	(((regs)->pstate & (PSR_MODE32_BIT | PSR_MODE_MASK)) == \
 	 (PSR_MODE32_BIT | PSR_MODE_EL0t))
 
+#define compat_user_mode(regs) a32_user_mode(regs)
+
 #define processor_mode(regs) \
 	((regs)->pstate & PSR_MODE_MASK)
 
